lib/adapters/serializers/csv.js

```diff
@@ -5,50 +5,107 @@ var base = require('./base');
 var csv = require('csv-write-stream');
 var errors = require('../../errors');
 var values = require('../../runtime/values');
+var readline = require('readline');
 
 var CSVSerializer = base.extend({
 
+    initialize: function(stream, options) {
+        options = options  || {};
+
+        if (options.append && !options.input) {
+            throw errors.runtimeError('RT-INTERNAL-ERROR', {
+                error: 'input stream must be provided when appending'
+            });
+        }
+
+        this.append = options.append;
+        this.input = options.input;
+        this.base = Promise.resolve();
+    },
+
     write: function(points) {
-        var self = this;
+        if (this.append && !this.readHeaders) {
```

Comment:
Can we set `this.base` to the headers-reading promise in the constructor if `this.append` to avoid the slightly awkward `readHeaders` thing?